src/main/java/wromaciej/hvac_sim/solver/externals/ChannelSolver.java

```diff
@@ -0,0 +1,121 @@
+
+import static wromaciej.hvac_sim.solver.result.SolverResultType.SOLVED;
+
+@Service
+public class ChannelSolver implements ExternalSolver<Channel<? extends MatterStream>> {
+
+    private JunctionSolver junctionSolver;
+
+    public ChannelSolver(JunctionSolver junctionSolver) {
+        this.junctionSolver = junctionSolver;
+    }
+
+    public Junction channelToMassFlowJunction(Channel toSolve) {
+        List<ParameterWithDirection> parametersWithDirections = new ArrayList<>();
+        parametersWithDirections.add(new ParameterWithDirection(toSolve.getInletStream().getMassFlow(), BondDirection.INLET));
+        parametersWithDirections.add(new ParameterWithDirection(toSolve.getOutletStream().getMassFlow(), BondDirection.OUTLET));
+        parametersWithDirections.add(toSolve.getExtraMassFlow());
+        return new Junction(parametersWithDirections, junctionSolver);
+    }
+
+    public Junction channelToPressureJunction(Channel toSolve) {
+        List<ParameterWithDirection> parametersWithDirections = new ArrayList<>();
+        Parameter<Pressure> inletPressureDefinition = new Parameter(ParameterType.PRESSURE, SI.PASCAL);
+        Parameter<Pressure> outletPressureDefinition = new Parameter(ParameterType.PRESSURE, SI.PASCAL);;
+        if (toSolve.getInletStream().getSpecificParameters().getAbsolutePressure().isDefined()) {
+            inletPressureDefinition = toSolve.getInletStream().getSpecificParameters().getAbsolutePressure();
+        } else {
+            toSolve.getInletStream().getSpecificParameters().getMatterDefinition().addParameter(inletPressureDefinition);
+        }
+        if (toSolve.getOutletStream().getSpecificParameters().getAbsolutePressure().isDefined()) {
+            outletPressureDefinition = toSolve.getOutletStream().getSpecificParameters().getAbsolutePressure();
+        } else {
+            toSolve.getOutletStream().getSpecificParameters().getMatterDefinition().addParameter(outletPressureDefinition);
+        }
+        parametersWithDirections.add(new ParameterWithDirection(inletPressureDefinition, BondDirection.INLET));
+        parametersWithDirections.add(new ParameterWithDirection(outletPressureDefinition, BondDirection.OUTLET));
+        parametersWithDirections.add(new ParameterWithDirection(toSolve.getPressureDrop(), BondDirection.OUTLET));
+        return new Junction(parametersWithDirections, junctionSolver);
+    }
+
+    public Junction channelToEnergyJunction(Channel toSolve) {
+        List<ParameterWithDirection> parametersWithDirections = new ArrayList<>();
+        ParameterWithDirection specificEnthalpyDifference;
+        Parameter<SpecificEnthalpy> inletEnthalpyDefinition = new Parameter(ParameterType.SPECIFIC_ENTHALPY, SI.JOULE.divide(SI.KILOGRAM));
+        Parameter<SpecificEnthalpy> outletEnthalpyDefinition = new Parameter(ParameterType.SPECIFIC_ENTHALPY, SI.JOULE.divide(SI.KILOGRAM));
+        if (toSolve.getInletStream().getSpecificParameters().getSpecificEnthalpy().isDefined()) {
```

Comment:
that method is long as shit. divide it into methods with meaningful names, so you could read a method like a book.